### ISSUE TITLE
[PROPOSAL] generate docs for structs

### ISSUE THREAD

**Issue by dkuku** [elixir-lang/ex_doc#1819]
I generated docs in one of my pet projects and spotted that pages for modules with only struct definitions are completely empty.
I checked how it behaves with ecto schemas and it's the same. It would be good to generate docs with the defined fields even when there is no `@type` in this module. I can have a look into that.

**Comment by josevalim** [comment]
There are no plans for this at the moment, you would have to explicitly document which fields of the struct you believe are public. If the set-theoretic types work on Elixir goes forward, the goal is to have a more expressive API for defining structs, including docs per field and types, then we will integrate that into ExDoc too. :) Thank you.

**Comment by Munksgaard** [comment]
I'd be interested in this feature as well!

@josevalim: I'm curious if you still think that the set-theoretic types work will enable features such as this, or have your thoughts changed?

**Comment by josevalim** [comment]
Yes, the plan is to introduce a new API for defining structs, probably in Elixir v1.20.

**Comment by Munksgaard** [comment]
Exciting! I'm looking forward to learning more, when the time comes. Thanks for the quick answer!
